src/utils.rs

```diff
@@ -1,18 +1,15 @@
-use bcrypt_pbkdf::bcrypt_pbkdf;
+use crate::viewing_key::VIEWING_KEY_SIZE;
+use sha2::{Digest, Sha256};
+use std::convert::TryInto;
 use subtle::ConstantTimeEq;
 
-// 5 rounds == ~300M gas (doesn't work with query) - creation/validation takes ~1.5s
-// 2 rounds == ~120M gas (works with query) - creation/validation takes ~1s
-const DEFAULT_COST: u32 = 2;
-const OUTPUT_SIZE: usize = 24;
-const SALT: &[u8] = b"bestspiceintheEU";
-
 pub fn ct_slice_compare(s1: &[u8], s2: &[u8]) -> bool {
     bool::from(s1.ct_eq(s2))
 }
 
-pub fn create_hashed_password(s1: &str) -> [u8; OUTPUT_SIZE] {
-    let mut output = [0u8; OUTPUT_SIZE];
-    let _ = bcrypt_pbkdf(s1, SALT, DEFAULT_COST, &mut output);
-    output
+pub fn create_hashed_password(s1: &str) -> [u8; VIEWING_KEY_SIZE] {
+    Sha256::digest(s1.as_bytes())
+        .as_slice()
+        .try_into()
+        .expect("Wrong password length")
```

Comment:
Users will not see the message in this panic. Is this something we want to return an error for, or is it ok to panic?

Reply:
It's ok to panic because this can't happen since `VIEWING_KEY_SIZE` is set to 32

Reply:
ah, cool - and Sha256 obviously returns that many bytes too.